### ISSUE TITLE
OCR Corpora Early 2025 Publication Thread

### ISSUE THREAD

**Issue by ctschroeder** [CopticScriptorium/OCR#3]
### OCR publication thread for early 2025

**Do not close this issue until all checkboxes below are complete or have been rescheduled:**

## List of corpora:

In [Processed OCR folder](https://github.com/CopticScriptorium/OCR/tree/main/Processed%20OCR) (need chapter divisions + sentence splitting+full automatic NLP processing like the bible corpora)
- Giron Legendes (11 docs)
    - chapter divisions should follow
    - [x] Daughters Zenon 132
       - [x] chapter divisions added/checked
       - [x] pb xml:id numbers corrected
       - [x] lb n tags changed to ed_line n
       - [x] ed_pg n tags changed to ed_page n
       - [x] pb xml:id n tags changed to pb xml:id
       - [x] metadata updated
       - [x] final validation at [https://gucorpling.org/gitdox/validate_sgml.py](https://gucorpling.org/gitdox/validate_sgml.py)
       - [x] this document should be divided into 3: 1) BNF 132.1 ff. 19-20, 2) Leiden fragment, 3) BNF 132.1 f 21
    - [x] Daughters Zenon 78
       - [x] chapter divisions added/checked
       - [x] pb xml:id numbers corrected
       - [x] lb n tags changed to ed_line n
       - [x] ed_pg n tags changed to ed_page n
       - [x] pb xml:id n tags changed to pb xml:id
       - [x] metadata updated
       - [x] final validation at [https://gucorpling.org/gitdox/validate_sgml.py](https://gucorpling.org/gitdox/validate_sgml.py)
    - [x] Daughters Zenon Crawford
       - [x] chapter divisions added/checked
       - [x] pb xml:id numbers corrected
       - [x] lb n tags changed to ed_line n
       - [x] ed_pg n tags changed to ed_page n
       - [x] pb xml:id n tags changed to pb xml:id
       - [x] metadata updated
       - [x] final validation at [https://gucorpling.org/gitdox/validate_sgml.py](https://gucorpling.org/gitdox/validate_sgml.py)
    - [x] Emperors Daughter 1
       - [x] chapter divisions added/checked
       - [x] lb n tags changed to ed_line n
       - [x] ed_pg n tags changed to ed_page n
       - [x] pb xml:id n tags changed to pb xml:id
       - [x] metadata updated
       - [x] final validation at [https://gucorpling.org/gitdox/validate_sgml.py](https://gucorpling.org/gitdox/validate_sgml.py)
    - [x] Emperors Daughter 2
       - [x] chapter divisions added/checked
       - [x] lb n tags changed to ed_line n
       - [x] ed_pg n tags changed to ed_page n
       - [x] pb xml:id n tags changed to pb xml:id
       - [x] metadata updated
       - [x] final validation at [https://gucorpling.org/gitdox/validate_sgml.py](https://gucorpling.org/gitdox/validate_sgml.py)
    - [x] Emperors Daughter 3
       - [x] chapter divisions added/checked
       - [x] lb n tags changed to ed_line n
       - [x] ed_pg n tags changed to ed_page n
       - [x] pb xml:id n tags changed to pb xml:id
       - [x] metadata updated
       - [x] final validation at [https://gucorpling.org/gitdox/validate_sgml.py](https://gucorpling.org/gitdox/validate_sgml.py)
    - [x] Emperors Daughter 4
       - [x] chapter divisions added/checked
       - [x] lb n tags changed to ed_line n
       - [x] ed_pg n tags changed to ed_page n
       - [x] pb xml:id n tags changed to pb xml:id
       - [x] metadata updated
       - [x] final validation at [https://gucorpling.org/gitdox/validate_sgml.py](https://gucorpling.org/gitdox/validate_sgml.py)
    - [x] Eve Serpent
       - [x] chapter divisions added/checked
       - [x] lb n tags changed to ed_line n
       - [x] ed_pg n tags changed to ed_page n
       - [x] pb xml:id n tags changed to pb xml:id
       - [x] metadata updated
       - [x] final validation at [https://gucorpling.org/gitdox/validate_sgml.py](https://gucorpling.org/gitdox/validate_sgml.py)
    - [x] Marina BNF
       - [x] chapter divisions added/checked
       - [x] lb n tags changed to ed_line n
       - [x] ed_pg n tags changed to ed_page n
       - [x] pb xml:id n tags changed to pb xml:id
       - [x] metadata updated
       - [x] final validation at [https://gucorpling.org/gitdox/validate_sgml.py](https://gucorpling.org/gitdox/validate_sgml.py)
    - [x] Marina Clarendon
       - [x] chapter divisions added/checked
       - [x] lb n tags changed to ed_line n
       - [x] ed_pg n tags changed to ed_page n
       - [x] pb xml:id n tags changed to pb xml:id
       - [x] metadata updated
       - [x] final validation at [https://gucorpling.org/gitdox/validate_sgml.py](https://gucorpling.org/gitdox/validate_sgml.py)
    - [x] Sacrifice Abraham
       - [x] chapter divisions added/checked
       - [x] lb n tags changed to ed_line n
       - [x] ed_pg n tags changed to ed_page n
       - [x] pb xml:id n tags changed to pb xml:id
       - [x] metadata updated
       - [x] final validation at [https://gucorpling.org/gitdox/validate_sgml.py](https://gucorpling.org/gitdox/validate_sgml.py)

**all documents above need to be moved to https://github.com/CopticScriptorium/auto-corpora when done**

In GitDox
- [ ] timothy.alexandria corpus prayer.athanasius
    - [ ] translation span (work by E. Sturgis via @amir-zeldes in progress)
    - [ ] entities and identities
    - [ ] metadata

- [ ] apocalypse.paul (2)
      - [ ] corpus name needed
      - [ ] other metadata updated
      - possibly error in data -- translation on p. 1043 begins with folio 24a but OCR coptic begins in the middle of folio 6a p. 533

- [ ] pscyril.alexandria
    - [x] On Mary still in XML mode (auto tagging?)
    - [ ] entities and identities

 - [ ] pscyril.jerusalem
    - [ ] on the cross
        - [ ] needs corpus name
        - [ ] metadata updated
        - [ ] chapter & verse need to be updated in spreadsheet based on open tags in XML
    - [ ] on Mary
        - [ ] needs corpus name
        - [ ] metadata updated
        - [ ] chapter & verse need to be updated in spreadsheet based on open tags in XML
- [ ] psepiphanius on Mary
    - [ ] translation span
    - [ ] entities and identities
    - [ ] metadata
 
- [ ] pschrysostom
    - [ ] translation span
    - [ ] entities and identities
    - [ ] metadata

- [ ] pscelestinus
- [ ] pstimothy.alex
- [ ] psote.psoi
- [ ] timothy.discourse

**Comment by ctschroeder** [comment]
I know we were going to put Giron in the "automatically processed straight from XML like the big bible corpora" bucket, but with the sentence splitting and chapter divisions required, I think it will be easier to do both in GitDox. Chapter divisions are easier to see in the spreadsheet mode, and you can use the entity mode to do the sentence-splitting. So I'm making the executive decision to put this stuff in GitDox. @amir-zeldes @EarlyCodices

**Comment by amir-zeldes** [comment]
Fine by me - 11 docs should still be manageable, I'd be more worried for really big stuff. Keep in mind that one potential advantage of "XML Bible mode" is that tokenization is done fresh each time as tokenizers improve, so if you're correcting any segmentation manually, GD can make more sense (or you could introduce pipes in the XML repo). If you want fresh re-tokenization each release, you should *not* put things in GD - that would be the other main reason not to choose it.

**Comment by ctschroeder** [comment]
I don't think we are checking segmentation manually, but this is a good point. We have lots to do so I'll ponder this. They already have pipes from Lydia's work. I found when I was working on other documents for the full-auto-from-xml mode, validation was a real pain for reasons I don't fully remember and probably repressed.

**Comment by ctschroeder** [comment]
Daughers of Zeno(n) is actually Life of Hilaria. Doc titles updated. @amir-zeldes can you confirm that we only need chapter not sentence divisions for the automatically processed docs? thanks!

**Comment by amir-zeldes** [comment]
Yes, that's right - if there are chapter tags I can automatically predict sentences within those (I just need to remember to switch that option on, but unless I'm very distracted I think I'll do it the right way 😅)

**Comment by ctschroeder** [comment]
Great. Thanks @amir-zeldes! I think I will leave the documents in _with pipes_ but remove the pipes for the auto folder. I'll validate everything and fix the metadata here and then remove the pipes when copying over.

**Comment by amir-zeldes** [comment]
You can but you don't have to - I have a setting that auto-removes them if I know a document needs re-tokenizing (which is most of auto-corpora)

**Comment by ctschroeder** [comment]
@amir-zeldes what do you do about identities for these? (I assume entities=automatic)

**Comment by ctschroeder** [comment]
Note to self: Discovered that in Life of Hilaria ("Daughters of Zeno") Giron inserted text from aLeiden ms between folios 20 and 21. The text is continuous and in the proper place, but according to our guidelines this document should be divided into 3: 1) BNF 132.1 ff. 19-20, 2) Leiden fragment, 3) BNF 132.1 f 21.

**Comment by amir-zeldes** [comment]
> @amir-zeldes what do you do about identities for these? (I assume entities=automatic)

I hadn't realized this, but as of now the auto-corpora online appear to have automatic entities and zero predicted identities, but also no metadata for entity or identity quality... We should probably change that. Adding the metadata is clear, but do we want auto identities in these corpora? 

The pro would be having the annotations + uniformity, the con is that errors will also contaminate the people and places pages on CTS... To be fair, quality is somewhat decent, but some errors are inevitable in NLP.

**Comment by ctschroeder** [comment]
I think identity predictions are fine but the bible corpora might be a problem re errors, since it is generally messier than the others, and errors in bound groups and segmentation multiply the later errors.
I will add entities=automatic to the OCR'd corpora metadata. Is there anything else I should do?

**Comment by ctschroeder** [comment]
@amir-zeldes the "Emperor's Daughter" work is attributed to Theodosius of Alexandria. Not clear he wrote it. We already have a corpus for theodosius.alexandria which contains the encomium for the archangel Michael. Should I add Emperor's Daughter to the existing corpus or should I make a separate one? I had it in a life.empdaughter corpus before I realized there was already a theodosius one. Likewise I have to figure out the urn -- in the "lives" bunch/namespace or theodosius

**Comment by amir-zeldes** [comment]
TBH this is the sort of question I usually ask you 😅 I think if this falls into the category of author-based corpora that's fine. If you think people would look for it elsewhere/would expect it to be standalone since it fits more into the "lives" pattern, then we can also have a dedicated corpus - whatever you think is best!

**Comment by ctschroeder** [comment]
Yeah I thought you might say that! I got decision fatigue dealing with these fragments from Giron

**Comment by amir-zeldes** [comment]
Is it a "life"?

**Comment by ctschroeder** [comment]
@amir-zeldes it's v fragmentary! Four small fragments survive. Not sure there are any others known (PATHS doesn't list any). https://archive.org/details/lgendescoptesfr00revigoog/page/n82/mode/2up
For the size and fragment alone I was thinking of putting it with theodosius.alexandria but yeah, it would get lost, but but I don't think anyone is looking for it LOL

**Comment by amir-zeldes** [comment]
I see... it is a bit small to be standalone, but I suppose it has nothing to do with that long Encomium. So yeah, maybe I'd lean separate corpus, the other one is all about that Encomium on Michael.

**Comment by ctschroeder** [comment]
arguments for in theodosisus corpus:
- it's small -- making its own corpus clutters ANNIS
- it's historically attributed to theodosius
- is it a life? hmmm

Against
- not sure it's by theodosius
- hidden
- no connection to the Michael piece in terms of manuscript distribution (diff codices, found in different places [wm & edfu]

**Comment by ctschroeder** [comment]
Other option is to have a corpus of fragmentary lives and narratives. Giron has at least two others of these in the dialogue between eve and the serpent and on the sacrifice of abraham

**Comment by amir-zeldes** [comment]
The more I think about it, the less I like it with the Encomium. Fragment corpus or standalone could both work IMO, but for fragment corpus you'd want at least one other work in the initial release, otherwise the corpus name makes no sense.

**Comment by ctschroeder** [comment]
oh yes there will at least be three in there -- eve & serpent, sacr abraham, and empdaughter.  Now the urns - anythoughts about the namespaces? I know you like it when the first thing after copticLit namespace is the same in a corpus

**Comment by amir-zeldes** [comment]
Yes, that would be convenient. So what do they have in common, that they are fragments of narratives? So maybe "narrfrag" or something? Can there by a separator of some kind inside the namespace part?

**Comment by ctschroeder** [comment]
yeah no problem, like besa.aphthonia and whatnot

**Comment by ctschroeder** [comment]
Ready for publication @amir-zeldes: 
life.hilaria
life.chrysostom
life.empdaughter

**Comment by ctschroeder** [comment]
Correction -- don't do life.empdaughter

**Comment by ctschroeder** [comment]
auto corpora from OCR are ready except some have horizontal metadata and some have vertical. @amir-zeldes do you want them regularized?

**Comment by amir-zeldes** [comment]
Thanks! Metadata should always be vertical (I assume you mean two columns, tab delim: `key <TAB> value <NEWLINE>`). Document metadata lives in a file called `<DOCNAME>_meta.tab` and corpus metadata in `<CORPUSNAME>_meta.tab`.
